dev/update_mlflow_versions.py

```diff
@@ -5,14 +5,49 @@
 import click
 from packaging.version import Version
 
+_PYTHON_VERSION_FILES = [
+    Path("mlflow", "version.py"),
+]
+
+_PYPROJECT_TOML_FILES = [
+    Path("pyproject.toml"),
+    Path("pyproject.skinny.toml"),
+]
+
+_JAVA_VERSION_FILES = Path("mlflow", "java").rglob("*.java")
+
+_JAVA_POM_XML_FILES = Path("mlflow", "java").rglob("*.xml")
+
+_JS_VERSION_FILES = [
+    Path(
+        "mlflow",
+        "server",
+        "js",
+        "src",
+        "common",
+        "constants.tsx",
+    )
+]
+
+_R_VERSION_FILES = [Path("mlflow", "R", "mlflow", "DESCRIPTION")]
+
 
 def get_current_py_version() -> str:
     text = Path("mlflow", "version.py").read_text()
     return re.search(r'VERSION = "(.+)"', text).group(1)
 
 
-def replace_dev_suffix_with(version, repl):
-    return re.sub(r"\.dev0$", repl, version)
+def get_java_py_version_pattern(version: str) -> str:
+    version_without_suffix = replace_dev_or_rc_suffix_with(version, "")
+    return rf"{re.escape(version_without_suffix)}(-SNAPSHOT)?"
+
+
+def get_java_new_py_version(new_py_version: str) -> str:
+    return replace_dev_or_rc_suffix_with(new_py_version, "-SNAPSHOT")
+
+
+def replace_dev_or_rc_suffix_with(version, repl):
+    return re.sub(r"\.dev0$|rc\d+$", repl, version)
```

Comment:
this is the main change, the rest is kind of moving stuff around

Reply:
Can we use this https://packaging.pypa.io/en/latest/version.html#packaging.version.Version.base_version?